plugins/catalog-backend/src/ingestion/LocationReaders.ts

```diff
@@ -258,24 +258,31 @@ export class LocationReaders implements LocationReader {
     }
   }
 
-  private async readLocation(
-    location: LocationSpec,
-  ): Promise<LocationProcessorResult> {
-    let locationResult: LocationProcessorResult | undefined;
+  private async readLocation(location: LocationSpec): Promise<Buffer> {
+    let data: Buffer | undefined = undefined;
+    let error: Error | undefined = undefined;
 
     await this.handleLocation(
       {
         type: 'location',
         location,
         optional: false,
       },
-      r => (locationResult = r),
+      output => {
+        if (output.type === 'error' && !error) {
+          error = output.error;
+        } else if (output.type === 'data' && !data) {
```

Comment:
Probably throw if we get `data` twice?

Reply:
Sure, that makes sense

Reply:
Remember though both of yall that this is super temporary and will be replaced by an entirely dedicated read pipe later :)